static/src/javascripts/projects/commercial/modules/article-body-adverts-wide.js

```diff
@@ -0,0 +1,241 @@
+            return Promise.resolve(false);
+        }
+
+        boot();
+
+        if (config.page.hasInlineMerchandise) {
+            var im = addInlineMerchAd();
+            im.then(waitForMerch).then(addInlineAds);
+            return im;
+        }
+
+        addInlineAds();
+
+        return Promise.resolve(true);
+    }
+
+    function boot() {
+        bodyAds = 0;
+        isWide = detect.getBreakpoint() === 'wide';
+        replaceTopSlot = isMobile = !isWide && detect.isBreakpoint({ max: 'tablet' });
+        getSlotName = replaceTopSlot ? getSlotNameForMobile : getSlotNameForDesktop;
+    }
+
+    function getSlotNameForMobile() {
+        bodyAds += 1;
+        return bodyAds === 1 ? 'top-above-nav' : 'inline' + (bodyAds - 1);
+    }
+
+    function getSlotNameForDesktop() {
+        bodyAds += 1;
+        return 'inline' + bodyAds;
+    }
+
+    function getRules(isMerch) {
+        var prevSlot;
+        var rules = {
+            bodySelector: '.js-article__body',
+            slotSelector: ' > p',
+            minAbove: isMobile ? 300 : config.page.hasShowcaseMainElement ? 900 : 700,
+            minBelow: adSizes.mpu.height,
+            selectors: {
+                ' > h2': {minAbove: isMobile ? 100 : 0, minBelow: 250},
+                ' .ad-slot': {minAbove: 500, minBelow: 500},
+                ' > :not(p):not(h2):not(.ad-slot)': {minAbove: 35, minBelow: 400}
+            },
+            filter: function(slot) {
+                if (!prevSlot || Math.abs(slot.top - prevSlot.top) - adSizes.mpu.height >= rules.selectors[' .ad-slot'].minBelow) {
+                    prevSlot = slot;
+                    return true;
+                }
+                return false;
+            }
+        };
+
+        if (!isMerch && isWide) {
+            rules.minBelow = 100;
+            rules.selectors = {
+                ' .ad-slot': { minAbove: 500, minBelow: 500 }
+            };
+        }
+
+        return rules;
+    }
+
+    function getInlineMerchRules() {
+        var inlineMerchRules = getRules(true);
+        inlineMerchRules.minAbove = 300;
+        inlineMerchRules.selectors[' > h2'].minAbove = 100;
+        inlineMerchRules.selectors[' > :not(p):not(h2):not(.ad-slot)'].minAbove = 200;
+        return inlineMerchRules;
+    }
+
+    function getLongArticleRules() {
+        var longArticleRules = getRules();
+        longArticleRules.selectors[' .ad-slot'].minAbove =
+        longArticleRules.selectors[' .ad-slot'].minBelow = Math.max(500, detect.getViewport().height);
+        return longArticleRules;
+    }
+
+    // Add new ads while there is still space
+    function addArticleAds(count, rules) {
+        return spaceFiller.fillSpace(rules, insertInlineAds, {
+            waitForImages: true,
+            waitForLinks: true,
+            waitForInteractives: true
+        });
+
+        function insertInlineAds(paras) {
+            var slots = paras
+            .slice(0, Math.min(paras.length, count))
```

Comment:
Ha, kind of 😅  C-like languages already have too much indentation, I feel like the `.` is enough cue to notice it is a continuation of the previous line.